code/modules/reagents/reagent_dispenser.dm

```diff
@@ -202,20 +202,33 @@
 	..()
 	reagents.add_reagent("condensedcapsaicin",1000)
 
-
 /obj/structure/reagent_dispensers/water_cooler
-	name = "Water-Cooler"
-	desc = "A machine that dispenses water to drink"
+	name = "water cooler"
+	desc = "A machine that dispenses water to drink."
 	amount_per_transfer_from_this = 5
-	icon = 'icons/obj/vending.dmi'
+	icon = 'icons/goonstation/objects/objects.dmi'
 	icon_state = "water_cooler"
 	possible_transfer_amounts = null
 	anchored = 1
+	var/number_of_cups = 15
+
 
 /obj/structure/reagent_dispensers/water_cooler/New()
 	..()
 	reagents.add_reagent("water",500)
 
+/obj/structure/reagent_dispensers/water_cooler/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>And has [number_of_cups] cup[number_of_cups == 1 ? "" : "s"] left in the holder.</span>")
+
+/obj/structure/reagent_dispensers/water_cooler/attack_hand(mob/user as mob)

```

Comment:
I wonder if this could actually just directly store the water cups instead of creating them--that way you could load the cups back into the tower after you're done with them? Dunno if it's really worth it or not though. Glad the damn water coolers finally have cups with them though.